services/sts/src/main/java/software/amazon/awssdk/services/sts/internal/StsWebIdentityCredentialsProviderFactory.java

```diff
@@ -50,30 +50,39 @@
 public final class StsWebIdentityCredentialsProviderFactory implements WebIdentityTokenCredentialsProviderFactory {
 
     public AwsCredentialsProvider create(WebIdentityTokenCredentialProperties credentialProperties) {
-        return new StsWebIdentityCredentialsProvider(credentialProperties);
+        return new StsWebIdentityCredentialsProvider(credentialProperties, null);
+    }
+
+    public AwsCredentialsProvider create(WebIdentityTokenCredentialProperties credentialProperties,
+                                         StsClient stsClient) {
+        return new StsWebIdentityCredentialsProvider(credentialProperties, stsClient);
     }
 
     /**
      * A wrapper for a {@link StsAssumeRoleWithWebIdentityCredentialsProvider} that is returned by this factory when
      * {@link #create(WebIdentityTokenCredentialProperties)} is invoked. This wrapper is important because it ensures the parent
      * credentials provider is closed when the assume-role credentials provider is no longer needed.
+     * It creates default StsClient if StsClient is null or if StsClient is not passed while creating the instance.
      */
     private static final class StsWebIdentityCredentialsProvider implements AwsCredentialsProvider, SdkAutoCloseable {
```

Comment:
This feels like a copy of StsAssumeRoleWithWebIdentityTokenCredentialsProvider with extra construction logic. I don't see why this has to be a class.

Two questions:
1. Should we just copy the configuration logic into the StsWebIdentityCredentialsProviderFactory, and just use the StsAssumeRoleWithWebIdentityTokenCredentialsProvider?
2. If we allow customers to override the StsClient, won't that remove the StsRetryCondition that they need? Should we move that StsRetryCondition to be retried universally so that customers won't override it by accident?